src/common/tusb_fifo.h

```diff
@@ -80,6 +80,14 @@ typedef struct
 
 } tu_fifo_t;
 
+typedef struct
+{
+  uint16_t len_lin                      ; ///< linear length in item size
+  uint16_t len_wrap                     ; ///< wrapped length in item size
+  void * ptr_lin                        ; ///< linear part start pointer
+  void * ptr_wrap                       ; ///< wrapped part start pointer
+} tu_fifo_linear_wr_info;
```

Comment:
nice idea, It could simplify the application usage for these direct buffer access. Though the name is not quite right since it is used for both read/write. Please change it to `tu_fifo_buffer_info_t` . I think the member name should be changed as well to simply buffer1 and buffer2, the lin/wrap will cause user question for those that aren't familiar with fifo implementation. 

```C
typedef struct
{
  uint16_t len1                      ; ///< length in item size of first buffer
  uint16_t len2                     ; ///< length in item size of second buffer (wrapped)
  void * buffer1                        ; ///< buffer1 pointer
  void * buffer2                       ; ///< buffer2 pointer (wrapped)
} tu_fifo_buffer_info_t;
```

Reply:
Mh, the name `buffer_info` is a bit too non-speaking don't you mean? `linear_wr_info` stands for linear write/read info.

Reply:
Ah i get your idea from below! Ok i rename it!